components/Checkbox/CheckboxGroup.jsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { FieldGroup, ErrorMessage } from '../shared';
+import Checkbox from './Checkbox';
+
+const Group = styled(FieldGroup)`
+  position: relative;
+`;
+
+const ErrorLabel = styled(ErrorMessage)`
+  margin-left: -3px;
+`;
+
+ErrorLabel.displayName = 'ErrorLabel';
+
+class CheckboxGroup extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const { children, options } = this.props;
+    const childrenProps = React.Children.count(children)
+      ? React.Children.toArray(children).map(
+          ({ props: childProps }) => childProps,
+        )
+      : options;
+
+    const items = childrenProps.map(({ checked, name, value }) => ({
+      checked: Boolean(checked),
+      name,
+      value,
+    }));
+
+    this.state = { items };
+  }
+
+  _onChange = event => {
+    const {
+      target: { checked, name },
+    } = event;
+    const { onChange } = this.props;
+    const { items } = this.state;
+
+    const newItems = items.map(item =>
+      item.name === name ? { ...item, checked } : item,
+    );
+
+    onChange(event, newItems);
+
+    this.setState({ items: newItems });
+  };
+
+  render() {
+    const { children, error, options } = this.props;
+
+    const commonProps = {
```

Comment:
What do you think about making a Context for CheckGroup?

By doing in this way, you will not need [this](https://github.com/catho/quantum/pull/43/files#diff-1ecd5a117b918d083f9d8f6d0b81664eR69) and it will works when the children prop weren't be a Checkbox. For example, you could have a conditional group of checks like this:

```
<CheckGroup>
  <Checkbox name="foo" />
  {someCheck && 
    <div>
      <Checkbox name="bar" />
      <Checkbox name="baz" />
    </div>
  }
  <Checkbox name="foo" />
</CheckGroup>
```

Of course you will need to add the Context to Checkbox itself what is ok for me.

Reply:
Done ✅